File: hotel-app/src/main/java/hotelico/service/booking/application/hotel/domain/infrastructure/dto/HotelActivityDTO.java
// path: hotelico/service/booking/application/hotel/domain/infrastructure/dto/HotelActivityDTO.java

package hotelico.service.booking.application.hotel.domain.infrastructure.dto;

import eu.getsoftware.hotelico.clients.common.domain.domainIDs.ActivityDomainEntityId;
import eu.getsoftware.hotelico.clients.common.domain.domainIDs.CustomerDomainEntityId;
import eu.getsoftware.hotelico.clients.common.domain.domainIDs.HotelDomainEntityId;
import eu.getsoftware.hotelico.clients.common.dto.BasicDTO;
import eu.getsoftware.hotelico.clients.common.utils.AppConfigProperties;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@RequiredArgsConstructor
public class HotelActivityDTO implements BasicDTO<ActivityDomainEntityId> {

    private final ActivityDomainEntityId domainEntityId;
    private final boolean active = true;
    
    private HotelDomainEntityId hotelId;

    private final long initId = 0;

    private int dealDaysDuration = 0;
    private int likeCounter = 0;
    private int subscribeCounter = 0;
    private int hotelCustomerNumber = 0;
    private int otherActivityNumber = 0;

    /**
     * 0: no limits
     * -1: no deals allowed
     */
    private int limitDealNumber = 0;

    private long consistencyId = 0;
    private long validDealId = 0;
    private int orderIndex = 0;

    @Getter
    private boolean publishInWall = false;
    private boolean likedByMe = false;
    private boolean hidden = false;
    private boolean subscribeByMe = false;
    private boolean timeValid = true;
    private boolean dealAllowed = true;
    private boolean lastMinute = false;

    private CustomerDomainEntityId senderDomainId;
    private LocalDate validFrom;
    private LocalDate validTo;

    private String title = "";
    private String activityArea = "";
    private String hotelCity = "";
    private String hotelName = "";
    private String shortDescription = "";
    private String description = "";
    private boolean thirdPartyActivity = false;

    private String pictureUrl;
    private String previewPictureUrl;

    public String getValidToString() {
        return validTo != null ? AppConfigProperties.dateFormat.format(validTo) : "";
    }

    public String getValidFromString() {
        return validFrom != null ? AppConfigProperties.dateFormat.format(validFrom) : "";
    }

    public String getPictureUrl() {
        return AppConfigProperties.addHostPrefixOnDemand(pictureUrl);
    }

    public String getPreviewPictureUrl() {
        return previewPictureUrl != null ? previewPictureUrl : AppConfigProperties.PREVIEW_ACTIVITY_NOT_AVAILABLE_URL;
    }

    public void setSequenceId(long initId) {
    }
}

File: hotel-app/src/main/kotlin/eu/getsoftware/hotelico/service/booking/application/hotel/domain/infrastructure/dto/HotelActivityDTO.kt
package eu.getsoftware.hotelico.service.booking.application.hotel.domain.infrastructure.dto;

import eu.getsoftware.hotelico.clients.common.domain.domainIDs.ActivityDomainEntityId
import eu.getsoftware.hotelico.clients.common.domain.domainIDs.CustomerDomainEntityId
import eu.getsoftware.hotelico.clients.common.domain.domainIDs.HotelDomainEntityId
import eu.getsoftware.hotelico.clients.common.dto.BasicDTO
import eu.getsoftware.hotelico.clients.common.utils.AppConfigProperties
import java.time.LocalDate

data class HotelActivityDTO(var hotelId: HotelDomainEntityId): BasicDTO<ActivityDomainEntityId>()
{
    val initId: Long = 0
    
    var dealDaysDuration = 0

    var likeCounter = 0

    var subscribeCounter = 0

    var hotelCustomerNumber = 0

    var otherActivityNumber = 0

    /**
     * o: no limits
     * -1: no deals allowed
     */
    var limitDealNumber = 0

    var consistencyId = 0L

    var validDealId = 0L

    var orderIndex = 0
 
    var publishInWall = false

    var likedByMe = false

    var hidden = false

    var subscribeByMe = false

    var timeValid = true

    var dealAllowed = true

    var lastMinute = false
    
    var senderDomainId: CustomerDomainEntityId? = null

    var validFrom: LocalDate? = null

    var validTo: LocalDate? = null

    var title = ""

    var activityArea = ""

    var hotelCity = ""

    var hotelName = ""

    var shortDescription = ""

    var description = ""

    var thirdPartyActivity = false

    private var pictureUrl: String? = null
    private var previewPictureUrl: String? = null

    fun isActiveBoolean(): Boolean {
        return active
    }

    fun getValidToString(): String {
        return AppConfigProperties.dateFormat.format(validTo)
    }

    fun getValidFromString(): String {
        return AppConfigProperties.dateFormat.format(validFrom)
    }

    fun getPictureUrl(): String? {
        return AppConfigProperties.addHostPrefixOnDemand(pictureUrl)
    }
    
    fun setPictureUrl(pictureUrl: String){
        this.pictureUrl = pictureUrl
    }
    
    fun getPreviewPictureUrl(): String {
        return previewPictureUrl ?: AppConfigProperties.PREVIEW_ACTIVITY_NOT_AVAILABLE_URL
    }

    fun setPreviewPictureUrl(previewPictureUrl: String) {
        this.previewPictureUrl = previewPictureUrl
    }
    
    constructor(param: Long) : this(HotelDomainEntityId(param.toString()))
}
